copernicus_ds_app.module

```diff
@@ -189,4 +190,52 @@ function copernicus_ds_app_preprocess_node(&$variables) {
   if ($load_ds_toolkit) {
     $variables['#attached']['library'][] = 'copernicus_ds_app/ds_app';
   }
+
 }
+
+function copernicus_ds_app_cron() {
+  $state = \Drupal::state();
+  $last_run = $state->get('copernicus_ds_app.weekly_update_last_run', 0);
+  $now = \Drupal::time()->getRequestTime();
+  $one_week_in_seconds = 7 * 24 * 60 * 60;
+  if (($now - $last_run) < $one_week_in_seconds) {
+    return;
+  }
+  $state->set('copernicus_ds_app.weekly_update_last_run', $now);
+  $paragraph_storage = \Drupal::entityTypeManager()->getStorage('paragraph');
+  $paragraph_ids = $paragraph_storage->getQuery()
+    ->accessCheck(TRUE)
+    ->condition('type', 'embed_data_store_app')
+    ->execute();
+
+  $http_client = \Drupal::httpClient();
+
+  foreach ($paragraph_ids as $pId) {
+    /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+    $paragraph = Paragraph::load($pId);
+    if (!$paragraph) {
+      continue;
+    }
+
+    if (!$paragraph->hasField('field_ds_app_link') || $paragraph->get('field_ds_app_link')->isEmpty()) {
+      continue;
+    }
+    $url = $paragraph->get('field_ds_app_link')->uri ?? $paragraph->get('field_ds_app_link')->value;
+    if (empty($url)) {
+      continue;
+    }
+
+    try {
+      $response = $http_client->get($url);
+      $json_string = $response->getBody()->getContents();
```

Comment:
I don't like you copy pasted the code here from above (DRY principle). Is it possible to create a single functions that makes the request and also handles errors?